saleor/graphql/product/mutations/products.py

```diff
@@ -1162,142 +1163,168 @@ def perform_mutation(cls, _root, info, **data):
         return response
 
 
-class ProductImageCreateInput(graphene.InputObjectType):
-    alt = graphene.String(description="Alt text for an image.")
+class ProductMediaCreateInput(graphene.InputObjectType):
+    alt = graphene.String(description="Alt text for a product media.")
     image = Upload(
-        required=True, description="Represents an image file in a multipart request."
+        required=False, description="Represents an image file in a multipart request."
     )
     product = graphene.ID(
         required=True, description="ID of an product.", name="product"
     )
+    video_url = graphene.String(required=False, description="Represents a video URL.")
 
 
-class ProductImageCreate(BaseMutation):
+class ProductMediaCreate(BaseMutation):
     product = graphene.Field(Product)
-    image = graphene.Field(ProductImage)
+    media = graphene.Field(ProductMedia)
 
     class Arguments:
-        input = ProductImageCreateInput(
-            required=True, description="Fields required to create a product image."
+        input = ProductMediaCreateInput(
+            required=True, description="Fields required to create a product media."
         )
 
     class Meta:
         description = (
-            "Create a product image. This mutation must be sent as a `multipart` "
-            "request. More detailed specs of the upload format can be found here: "
+            "Create a media object (image or video URL) associated with product. "
+            "For image, this mutation must be sent as a `multipart` request. "
+            "More detailed specs of the upload format can be found here: "
             "https://github.com/jaydenseric/graphql-multipart-request-spec"
         )
         permissions = (ProductPermissions.MANAGE_PRODUCTS,)
         error_type_class = ProductError
         error_type_field = "product_errors"
 
+    @classmethod
+    def validate_input(cls, data):
+        image = data.get("image")
+        video_url = data.get("video_url")
+
+        if not image and not video_url:
+            raise ValidationError(
+                {"input": ValidationError("Image or video URL is required.")}
```

Comment:
ValidationError should also provide an error code, in this case `ProductErrorCode.REQUIRED`. See how it is used in other validation errors.